test/autoproj/cli/test_daemon.rb

```diff
@@ -21,6 +21,7 @@ module CLI
                     url: "git@github.com:rock-core/buildconf"
                 )
                 @cli = Daemon.new(ws)
+                flexmock(cli).should_receive(:restart_and_update).never.by_default
```

Comment:
What's `by_default` for?

Reply:
It mocks the method, but has a lower priority than a non-by_default mock, thus allowing to override the spec later.

Reply:
Hmm.. Didn't know that. Useful.